torchgen/model.py

```diff
@@ -1723,6 +1728,31 @@ def symint_to_int(self) -> "Type":
     def is_list_like(self) -> Optional["ListType"]:
         return self.elem.is_list_like()
 
+# A type representing a PyTorch custom class
+@dataclass(frozen=True)
+class CustomClassType(Type):
+    class_name: str
+
+    def __str__(self) -> str:
+        return self.class_name
```

Comment:
nit: it might be clearer to store the prefix and the actual class name as separate fields on `CustomClassType`, since the `__torch__.torch.classes` bit technically isn't part of the class name.

Reply:
yep. I'll store the real class name skipping the prefix since the parsing logic assumes such a prefix anyways.